### PR TITLE
Feature/UI wallet

### PR DESCRIPTION
**Pull request by GrigoriGrigoryanBytepitch**

### CODE REVIEW

**Review by Jorge-Lopes — APPROVED** [review]
LGTM

File: ui/package.json
[the remark this answers is not in the document]
Comment (Jorge-Lopes) [reply ui/package.json:1]: Strange, I didn't noticed before

[the remark this answers is not in the document]
Comment (Jorge-Lopes) [reply ui/package.json:1]: Fixed

**Review by alexanderem49 — APPROVED** [review]
LGTM

File: ui/src/Trade/ExchangeInterface.jsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import { useStore } from '../store/store.js';
+import { XCircleIcon } from '@heroicons/react/24/outline/index.js';
+import { buyOffer, makeGenericOnStatusUpdate } from '../utils/makeOrder.js';
+
+export default function ExchangeInterface() {
+  const [inputValue, setInputValue] = useState('');
+  const [outputValue, setOutputValue] = useState('');
+  const [firstLabel, setFirstLabel] = useState('Asset:');
+  const [secondLabel, setSecondLabel] = useState('Price:');
+  const [assetBrand, setAssetBrand] = useState('BLD');
```
Comment (anilhelvaci) [inline ui/src/Trade/ExchangeInterface.jsx:12]: I think this isn't up to date. Are you sure you pushed the latest version?

File: ui/src/Trade/ExchangeInterface.jsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  const [priceBrand, setPriceBrand] = useState('IST');
```
Comment (anilhelvaci) [inline ui/src/Trade/ExchangeInterface.jsx:13]: Ditto here

File: ui/src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      window.removeEventListener('hashchange', handleHashChange);
+    };
+  }, []);
+
+  return (
+    <ErrorBoundary>
+      <div>
+        <NavBar />
+        {hash === '#trade' && <Trade />}
+        {/*{hash === "#about" && <About />}*/}
```
Comment (anilhelvaci) [inline ui/src/App.jsx:35]: Better to delete these I believe

File: ui/src/utils/makeOrder.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+harden(buyOffer);
+harden(sellOffer);
+
+export const makeGenericOnStatusUpdate = (snackBarUpdater) => {
+  const onStatusChange = (args) => {
+    console.log({ args });
+    const { status, data } = args;
+
+    if (status === 'error') {
+      snackBarUpdater('error', 'Offer with error');
```
Comment (anilhelvaci) [inline ui/src/utils/makeOrder.js:46]: Do we have a snackbar?

File: ui/src/store/store.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  setOrders: (simpleExchange) => {
+    const buyOrders = [...simpleExchange.buys];
+    const sellOrders = [...simpleExchange.sells];
+    console.log('buyOrders: ', buyOrders);
+    console.log('sellOrders: ', sellOrders);
+    set({ buyOrders, sellOrders });
+  },
+  notifyUser: (severity, message) => {
+    set(() => ({
+      notifierState: { open: true, severity, message }
```
Comment (anilhelvaci) [inline ui/src/store/store.js:29]: I think this is for the snackbar. Do you think we should add it? @GrigoriGrigoryanBytepitch

**Review by anilhelvaci — CHANGES_REQUESTED** [review]

File: ui/src/store/store.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  setExchangedBrands: (vbankAssets) => {
+    const assetBrand = {};
+    const priceBrand = {};
+    console.log('vbankAssets: ', vbankAssets);
+    vbankAssets.forEach(([denom, assetInfo]) => {
+      if (denom === 'ubld') {
+        assetBrand[assetInfo.issuerName] = assetInfo;
+      } else if (denom === 'uist') {
+        priceBrand[assetInfo.issuerName] = assetInfo;
+      }
```
Comment (anilhelvaci) [inline ui/src/store/store.js:45]: I see that we're still getting the price and asset brand by querying the denoms.

File: ui/src/Trade/ExchangeInterface.jsx (hunk 1 above)
Comment (anilhelvaci) [reply ui/src/Trade/ExchangeInterface.jsx:11]: Ditto

File: ui/src/Trade/ExchangeInterface.jsx (hunk 2 above)
Comment (anilhelvaci) [reply ui/src/Trade/ExchangeInterface.jsx:12]: Ditto

**Pull request merged**